project-minimization-plugin/src/main/kotlin/org/plan/research/minimization/plugin/services/GitWrapperService.kt

```diff
@@ -0,0 +1,127 @@
+package org.plan.research.minimization.plugin.services
+
+import  org.plan.research.minimization.plugin.model.IJDDContext
+
+import com.intellij.openapi.application.readAction
+import com.intellij.openapi.components.Service
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.project.isProjectOrWorkspaceFile
+import com.intellij.openapi.util.io.FileUtil
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.openapi.vfs.toNioPathOrNull
+import org.eclipse.jgit.api.Git
+import org.eclipse.jgit.api.ResetCommand
+import org.eclipse.jgit.api.errors.NoHeadException
+
+import java.io.File
+import java.util.*
+
+import kotlin.io.path.pathString
+import kotlin.io.path.relativeTo
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+/**
+ * Service responsible for git operations within ProjectGitSnapshotManager.
+ */
+@Service(Service.Level.APP)
+class GitWrapperService {
+    private val importantFiles = setOf("modules.xml", "misc.xml", "libraries")
+
+    suspend fun commitChanges(context: IJDDContext): IJDDContext = context.apply {
+        commit(this.projectDir)
+        projectDir.refresh(false, true)
+    }
+
+    suspend fun resetChanges(context: IJDDContext) {
+        withContext(Dispatchers.IO) {
+            // git reset --hard
+            Git.open(context.projectDir.toNioPath().toFile()).apply {
+                reset().setMode(ResetCommand.ResetType.HARD).call()
+                clean().setCleanDirectories(true).call()
+                close()
+            }
+        }
+    }
+
+    private suspend fun commit(projectDir: VirtualFile) = withContext(Dispatchers.IO) {
+        Git.open(projectDir.toNioPath().toFile()).let {
+            projectDir.gitAdd(it) { file ->
+                isImportant(file, projectDir)
+            }
```

Comment:
That is, of course, discussable, and I just made this idea right now: but since on the all stages we manipulate only files from source roots, maybe we could just do a `git add` on initialization and after it just git all the roots?

Reply:
I guess we can refactor this a little.
1. Add a separate function for `gitAdd` with an argument that gives a function of "importance"
2. Do not add files in this function
3. Add function `initialize` which will help to remove/reinit .git, add "important" files and make an initial commit
4. Call initialization of this service in the cloning service on every new clone

Reply:
It will also help to delete duplicate code with `isImportant`

Reply:
Oh, we already have "initialize function", so we can just move add logic there (and add an importance function argument)